### PR TITLE
6.0 dynamic

### PR DESCRIPTION
**Pull request by lbrant1**
Add is_dynamic option to seeker to allow users to more easily integrate ajax and js changes on standard seekerview views.

### CODE REVIEW

File: seeker/facets.py [hunk 1]
```diff
@@ -473,11 +479,29 @@ def data(self, response, values=[], **kwargs):
...
+        self.format_validator = format_validator
         super(DateRangeFacet, self).__init__(field, **kwargs)
 
     def _get_filter_from_range_list(self, _range):
-        _range = super(DateRangeFacet, self)._get_filter_from_range_list(_range)
+        """
+        This helper function is designed to take a list of 2 values and build a range query.
+        """
+        if isinstance(_range, dict):
+            r = _range
```
Comment (ruttenb) [inline seeker/facets.py:492]: If this is a dict, do we need to validate the date format?

File: seeker/utils.py [hunk 2]
```diff
@@ -151,3 +155,12 @@ def convert_saved_search_to_search_object(saved_search):
...
     return search_object
+
+
+def validate_date_format(date_text, date_format):
+    try:
+        if date_text != datetime.strptime(date_text, date_format).strftime(date_format):
+            raise ValueError
+        return True
+    except ValueError:
+        return False
```
Comment (ruttenb) [inline seeker/utils.py:164]: I think we can shorten this a bit:

    try:
        return date_text == datetime.strptime(date_text, date_format).strftime(date_format)
    except ValueError:
        return False

File: seeker/views.py [hunk 3]
```diff
@@ -266,6 +287,11 @@ def required_display_fields(self):
     """
     The number of results to show per page.
     """
+    
+    page_size_options = []
+    """
+    If set allows user to set options for page size to be changed
+    """
```
Comment (ruttenb) [inline seeker/views.py:289]: Take a look at `AdvancedSeeker`.  Advanced seeker has `available_page_sizes`.  We should probably merge the two (i.e - Move available_page_sizes to SeekerView)

File: seeker/views.py [hunk 4]
```diff
@@ -397,6 +425,13 @@ def modify_context(self, context, request):
         """
         pass
 
+    def set_page_size(self):
+        ps = self.request.GET.get('page_size', '').strip()
+        try:
+            self.page_size = int(ps) if int(ps) > 0 and int(ps) in self.page_size_options else self.page_size
+        except ValueError:
+            pass
+
```
Comment (ruttenb) [inline seeker/views.py:429]: In what use case does a `ValueError` get thrown?  Also, I'm not a huge fan of updating self.page_size; I think this function should return the page size instead.  Something like this: 

```
def set_page_size(self):
    ps = int(self.request.GET.get('page_size', self.page_size))
    return ps if ps in self.page_size_options else self.page_size
```

Comment (ruttenb) [reply seeker/views.py:429]: I would also rename it to `get_page_size`

File: seeker/views.py [hunk 5]
```diff
@@ -757,11 +805,12 @@ def render(self):
...
         # Highlight fields.
         if self.highlight:
-            highlight_fields = self.highlight if isinstance(self.highlight, (list, tuple)) else [c.highlight for c in columns if c.visible and c.highlight]
+            highlight_fields = self.highlight if isinstance(self.highlight, (list, tuple)) else [c.highlight for c in columns if (c.visible or c.field in self.hidden_columns) and c.highlight]
             # NOTE: If the option to customize the tags (via pre_tags and post_tags) is added then the Column "render" function will need to be updated.
-            search = search.highlight(*highlight_fields, number_of_fragments=0).highlight_options(encoder=self.highlight_encoder)
+            search = search.highlight(*highlight_fields, number_of_fragments=self.number_of_fragments).highlight_options(encoder=self.highlight_encoder)
 
         # Calculate paging information.
+        self.set_page_size()
```
Comment (ruttenb) [inline seeker/views.py]: If you update the function to return a value, then you can update this to be `page_size = self.get_page_size()` and use page_size below.  This is similar to how AdvancedSeeker does it

File: seeker/views.py [hunk 6]
```diff
@@ -215,12 +220,23 @@ class SeekerView(View):
     """
     A list of field names to exclude when generating columns.
     """
+    
+    hidden_columns = []
+    """
+    A list of field names to hide from display in columns but still allows templates to access the field with highlighting
+    """
```
Comment (ruttenb) [inline seeker/views.py]: I'm not sure I completely understand the use case for this list?

File: seeker/views.py (hunk 4 above)
Comment (lbrant1) [reply seeker/views.py:429]: A ValueError could be thrown if the querystring value for ps is not an int.

File: seeker/views.py (hunk 6 above)
Comment (lbrant1) [reply seeker/views.py]: If there is a field column you want to exclude from a search view's display columns but have its highlighted contents available to use inside another column.

File: seeker/facets.py (hunk 1 above)
Comment (lbrant1) [reply seeker/facets.py:492]: Honestly not sure, this function is assigning values for ranges if it's a list. If the range is a dict, it already has those values assigned then shouldn't they have been validated there?

File: seeker/views.py (hunk 6 above)
Comment (ruttenb) [reply seeker/views.py]: I think `exclude` should work for that use case, right? - https://github.com/imsweb/django-seeker/blob/6.0/seeker/views.py#L214

Comment (lbrant1) [reply seeker/views.py]: No, if the field name is in the exclude list its column is never initialized in the get_columns function and is thus not available in the render / results.

File: seeker/facets.py (hunk 1 above)
Comment (ruttenb) [reply seeker/facets.py:492]: I'm good with leaving this as-is

**Pull request merged**